core/main/src/firebolt/handlers/advertising_rpc.rs

```diff
@@ -433,3 +412,66 @@ impl RippleRPCProvider<AdvertisingImpl> for AdvertisingRPCProvider {
         (AdvertisingImpl { state }).into_rpc()
     }
 }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::firebolt::handlers::advertising_rpc::AdvertisingImpl;
+    use ripple_sdk::{api::gateway::rpc_gateway_api::JsonRpcApiRequest, tokio};
+    use ripple_tdk::utils::test_utils::Mockable;
+    use serde::{Deserialize, Serialize};
+    use serde_json::json;
+
+    #[derive(Serialize, Deserialize, Clone, Debug)]
+    struct CallContextContainer {
+        pub ctx: Option<CallContext>,
+    }
+
+    fn merge(a: &mut Value, b: &Value) {
+        match (a, b) {
+            (&mut Value::Object(ref mut a), Value::Object(ref b)) => {
+                for (k, v) in b {
+                    merge(a.entry(k.clone()).or_insert(Value::Null), v);
+                }
+            }
+            (a, b) => {
+                *a = b.clone();
+            }
+        }
+    }
+
+    fn test_request(
+        method_name: String,
+        call_context: Option<CallContext>,
+        params: Option<Value>,
+    ) -> String {
+        let mut the_map = params.map_or(json!({}), |v| v);
+        merge(
+            &mut the_map,
+            &serde_json::json!(CallContextContainer { ctx: call_context }),
+        );
+        let v = serde_json::to_value(JsonRpcApiRequest {
+            jsonrpc: "2.0".to_string(),
+            id: Some(1),
+            method: method_name,
+            params: Some(the_map),
+        })
+        .unwrap();
+        serde_json::to_string(&v).unwrap()
+    }
+
+    #[tokio::test]
+    pub async fn test_app_bundle_id() {
```

Comment:
Adding this as a starting point for one possible approach to testing these handlers.  This does not work for any functions  that are integrated (anything that uses `StorageManager` directly in this case).